### DISCUSSION TITLE
Not working multitenancy example code

### DISCUSSION THREAD

**Discussion by medihack** [blitz-js/blitz#2570] [Q&A] [answered]
Hi!

I am trying to implement a multitenancy architecture [as described in the documentation](https://blitzjs.com/docs/multitenancy). Unfortunately, some of this code does not seem to work as described (some examples below). Is there a Github repo with some working example code?

Here some things that don't work:

- `isAuthorized: SimpleRolesIsAuthorized<Role>`: Is role still defined, or must it be `MembershipRole` or `GlobalRole`?
- `roles: Array<MembershipRole & GlobalRole>` gives me the type `never[]` with `MembershipRole` and `GlobalRole` as defined in example.
- `user.memberships[0]`: `User` type does not have a `memberships` attribute when returned from the `authenticateUser` function. I migrated the schema, regenerated the prisma model and restarted the dev server. Do I need something else to make the reverse reference appear?

**Comment by medihack** [comment]
Ok, I solved one part by including the memberships in the query with `await db.user.findFirst({ where: { email }, include: { memberships: true } })`. The roles type is still incorrect because of the wrong intersection operator I guess. Also `user.role` does not exist anymore in the multitenancy example schema, but is used in the example as `roles: [user.role, user.memberships[0].role]`. I also wonder why `GlobalRole` is part of the `Organization` model. Shouldn't it be part of the `User` model?

**Comment by flybayer** [comment, marked as answer]
I've since updated the docs to fix a few issues (could still be a bug). Can you compare with the latest doc page?

**Reply by medihack** [reply]
Ah yes, makes more sense now. Thank you! Just as a footnote, a little typo in the last code sample (`GloblRole`).
